airgun/views/contentview_new.py

```diff
@@ -56,3 +88,148 @@ def is_displayed(self):
     def after_fill(self, value):
         """Ensure 'Create content view' button is enabled after filling out the required fields"""
         self.submit.wait_displayed()
+
+
+class ContentViewEditView(BaseLoggedInView):
+    breadcrumb = BreadCrumb()
+    search = PF4Search()
+    title = Text("//h2[contains(., 'Publish) or contains(@id, 'pf-wizard-title-0')]")
+    actions = ActionsDropdown(
+        "//div[contains(@data-ouia-component-id, 'OUIA-Generated-Dropdown-2')]"
+    )
+    publish = PF4Button('cv-details-publish-button')
+    # not sure if this is needed
+    dialog = ConfirmationDialog()
+
+    @property
+    def is_displayed(self):
+        breadcrumb_loaded = self.browser.wait_for_element(self.breadcrumb, exception=False)
+        return (
+            breadcrumb_loaded
+            and len(self.breadcrumb.locations) <= 3
+            and self.breadcrumb.locations[0] == 'Content Views'
+            and self.breadcrumb.read() != 'New Content View'
+            and self.publish.is_displayed
+        )
+
+    @View.nested
+    class details(Tab):
+        TAB_LOCATOR = ParametrizedLocator('//a[contains(@href, "#/details")]')
+        name = EditableEntry(name='Name')
+        label = ReadOnlyEntry(name='Label')
+        type = ReadOnlyEntry(name='Composite?')
+        description = EditableEntry(name='Description')
+        # depSolv is maybe a conditionalswitch
+        solve_dependencies = Switch(name='solve_dependencies switch')
+        import_only = Switch(name='import_only_switch')
+
+    @View.nested
+    class versions(Tab):
+        TAB_LOCATOR = ParametrizedLocator('//a[contains(@href, "#/versions")]')
+        searchbox = PF4Search()
+        table = PatternflyTable(
+            component_id="content-view-versions-table",
+            column_widgets={
+                0: Checkbox(locator='.//input[@type="checkbox"]'),
+                'Version': Text('.//a'),
+                'Environments': Text('.//a'),
+                'Packages': Text('.//a'),
+                'Errata': Text('.//a'),
+                'Additional content': Text('.//a'),
+                'Description': Text('.//a'),
+                7: Dropdown(locator='.//div[contains(@class, "pf-c-dropdown")]'),
+            },
+        )
+        publishButton = PF4Button('cv-details-publish-button')
+
+        def search(self, version_name):
+            """Searches for content view version.
+            Searchbox can't search by version name, only by id, that's why in
+            case version name was passed, it's transformed into recognizable
+            value before filling, for example::
+                'Version 1.0' -> 'version = 1'
```

Comment:
Is this really the correct mapping? Can't version 3.1 have id 16?

Reply:
This is just talking about mapping "Version 22" to search term '22", since you can't actually search "Version 22"

Reply:
Updated language